util/dvsim/Deploy.py

```diff
@@ -405,6 +405,8 @@ def __init__(self, index, test, build_job, sim_cfg):
         self.test_obj = test
         self.index = index
         self.seed = RunTest.get_seed()
+        self.simulated_time = 0
+        self.job_runtime = 0
```

Comment:
IMO `job_runtime` applies to builds as well. I wonder if we should make this a `Launcher` class property instead?

Reply:
That is true :) We can move the time variables to Launcher class. But for display, shall we add one more column for `job_buildtime`?

Reply:
I don't think so - it will be the same value for all tests, so it's an overkill to add a column that displays the same value. Lets keep this value computed and ready to go - we can display it if necessary, later.

Reply:
Okay sounds good! That is what I was thinking too - most of the tests will share the same build time value.

Reply:
It may be worth maintaining and displaying this for `CompileOneShot` derivatives (lint, formal, synthesis etc), so IMO it should be a `Launcher` class property, which will make it applicable for all types of jobs.

Reply:
Huh? CompileSim and CompileOneShot are derived from Deploy... Each Deploy has a Launcher, so it can be in either base class. If it is already in the Deploy side, lets keep it.

Reply:
@matutem, I suggested `Launcher` class because you would be computing and maintaining the job elapsed time as you add support for the job timeout mechanism. If we are unable to infer the actual job runtime using the simulator provided stats, we could default to using this elapsed time that you compute. The `Deploy` abstraction represents "what it takes to deploy this build or run job" - so it mainly deals with generating the `sim.mk` make command. Once that is done, it lets the next stage, i.e. the `Launcher` class object to deal with the actual launching and running of the job. In my mind, how long the job ran i.e. `job_runtime` aligns better with the `Launcher` class.

Reply:
But how about `simulated_time` variable? If we also declare it inside the launcher class, does the launcher know if we are in `run` or `build`, `coverage` etc step?

Reply:
Yeah - simulated time could remain a Deploy (or better, RunTest) class property.

Reply:
I see, thanks for explaining! Then I should probably separate the `get_runtimes` function to `get_simulated_time` and `get_job_runtime`. Then we can call them separately to get each property!

Reply:
THere is no need to split. you can do something like this:
```python
    # In Launcher::_check_status()
    runtime, simtime = get_runtimes(reversed(lines))
    if self.deploy.target == "run":
        self.deploy.simtime = simtime
```

Reply:
Thanks Sri! Actually wonder if it is slightly easier to split the `get_runtimes` function.

1). OneShotCompile only has `job_runtime`, it will be easier for them to just add a `get_XXX_runtime` in the `sim_utils.py` file.
2). In the deploy target is `build`, we are only looking for `job_runtime`, then the function can just return once we found the `job_runtime`. No need to continue searching for `simulated_time` through the entire file.
3). It is easier to handle the exceptions during `build` target. We do not need to worry about the exceptions of not finding `simulated_time`.

This is just my thought :) What do you think?

Reply:
Okay sure. I agree with you.